presto-hive/src/test/java/com/facebook/presto/hive/TestAriaHiveDistributedQueries.java

```diff
@@ -428,6 +436,26 @@ public void testQueenOfTheNight()
                 "    AND p.availqty < 1000");
     }
 
+    @Test
+    public void TestConstantColumns()
```

Comment:
- test methods should start with lower case, e.g. testConstantColumns
- SQL keywords need to be capitalized, e.g. SELECT vs. select
- testConstantColumns name is confusing; it suggests that column value is constant, but it actually refers to a column missing from an ORC file due to schema evolution; I'd rename this test to testSchemaEvolution
- I'm concerned about testing aria scan using non-aria scan; if we do that, how would we remove non-aria scan once aria code path is solid; I suggest to use H2 for this test and I'm showing how this can be done below
- I'd move tests for non-deterministic filters into a separate method

```
    @Test
    public void testSchemaEvolution()
    {
        assertUpdate("CREATE TABLE test_schema_evolution AS SELECT * FROM nation", 25);
        assertUpdate("ALTER TABLE test_schema_evolution ADD COLUMN nation_plus_region BIGINT");
        assertUpdate("INSERT INTO test_schema_evolution SELECT *, nationkey + regionkey FROM nation", 25);
        assertUpdate("ALTER TABLE test_schema_evolution ADD COLUMN nation_minus_region BIGINT");
        assertUpdate("INSERT INTO test_schema_evolution SELECT *, nationkey + regionkey, nationkey - regionkey FROM nation", 25);

        String cte = "WITH test_schema_evolution AS (" +
                "SELECT *, null AS nation_plus_region, null AS nation_minus_region FROM nation " +
                "UNION ALL SELECT *, nationkey + regionkey, null FROM nation " +
                "UNION ALL SELECT *, nationkey + regionkey, nationkey - regionkey FROM nation)";

        assertQueryUsingH2Cte("SELECT * FROM test_schema_evolution", cte);
        assertQueryUsingH2Cte("SELECT * FROM test_schema_evolution WHERE nation_plus_region IS NULL", cte);
        assertQueryUsingH2Cte("SELECT * FROM test_schema_evolution WHERE nation_plus_region > 10", cte);
        assertQueryUsingH2Cte("SELECT * FROM test_schema_evolution WHERE nation_plus_region + 1 > 10", cte);
        assertQueryUsingH2Cte("SELECT * FROM test_schema_evolution WHERE nation_plus_region + nation_minus_region > 20", cte);

        assertUpdate("DROP TABLE test_schema_evolution");
    }

    private void assertQueryUsingH2Cte(String query, String cte)
    {
        assertQuery(ariaSession(), query, cte + " " + query);
    }

    @Test
    public void testNonDeterministicConstantFilters()
    {
        int count = computeActual(ariaSession(), "SELECT * FROM nation WHERE random(10) < 5").getRowCount();
        assertTrue(count > 0 && count < 25);

        count = computeActual(ariaSession(), "SELECT * FROM nation WHERE random(10) < 0").getRowCount();
        assertEquals(count, 0);

        count = computeActual(ariaSession(), "SELECT * FROM nation WHERE random(10) < 10").getRowCount();
        assertEquals(count, 25);
    }
```